From 4742e81159d5a54494f497a79834cf746f236681 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date: Thu, 24 Jan 2019 14:03:28 +0000
Subject: [PATCH 318/703] char: vcio: Fail probe if rpi_firmware is not found.

Device Tree is now the only supported config mechanism, therefore
uncomment the block of code that fails the probe if the
firmware node can't be found.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>
---
 drivers/char/broadcom/vcio.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/drivers/char/broadcom/vcio.c
+++ b/drivers/char/broadcom/vcio.c
@@ -126,10 +126,9 @@ static int __init vcio_init(void)
 
 	np = of_find_compatible_node(NULL, NULL,
 				     "raspberrypi,bcm2835-firmware");
-/* Uncomment this when we only boot with Device Tree
 	if (!of_device_is_available(np))
 		return -ENODEV;
-*/
+
 	vcio.fw = rpi_firmware_get(np);
 	if (!vcio.fw)
 		return -ENODEV;
